src/chains/ethereum/ethereum/tests/blockchain.test.ts

```diff
@@ -0,0 +1,114 @@
+      await blockchain.initialize(wallet.initialAccounts);
+      const common = blockchain.common;
+
+      // set up two transactions
+      const [from, to] = wallet.addresses;
+      const secretKey = wallet.unlockedAccounts.get(from);
+      const rpcTx: EIP1559FeeMarketRpcTransaction = {
+        from,
+        to,
+        gasLimit,
+        maxFeePerGas: "0xffffffff",
+        type: "0x2",
+        nonce: "0x0"
+      };
+      const transaction1 = TransactionFactory.fromRpc(rpcTx, common);
+      const transaction2 = TransactionFactory.fromRpc(
+        { ...rpcTx, nonce: "0x1" },
+        common
+      );
+
+      const transaction1Promise = blockchain.queueTransaction(
+        transaction1,
+        secretKey
+      );
+      const transaction2Promise = blockchain.queueTransaction(
+        transaction2,
+        secretKey
+      );
+      // we're in strict mode, so transactions are in the pool and not yet mined
+      const transactionHashes = await Promise.all([
+        transaction1Promise,
+        transaction2Promise
+      ]);
+
+      const timestamps: number[] = [];
+      let i = 0;
+      const startingBlockNumber =
+        blockchain.blocks.latest.header.number.toNumber();
+
+      // wait for our two transactions to be mined by waiting for the chain to
+      // emit the block events. verify that it's the correct transaction, and
+      // store the timestamp
+      const assertBlocks = new Promise((resolve, reject) => {
+        const off = blockchain.on("block", async (block: Block) => {
+          const transactions = block.getTransactions();
+          if (transactions.length === 0) return;
+          // a failed assertion inside of this promise doesn't fail the test,
+          // it just stops this promise early. so catch any failures, reject the
+          // promise, and use that rejection to fail the test
+          try {
+            const blockNumber = block.header.number.toString();
+            const expectedBlockNumber = startingBlockNumber + i + 1;
+            assert.equal(blockNumber, `0x${expectedBlockNumber.toString(16)}`);
+            assert.equal(transactions.length, 1);
+
+            const transaction = transactions[0];
+            assert.equal(
+              transaction.hash.toString(),
+              transactionHashes[i].toString()
+            );
+
+            timestamps.push(block.header.timestamp.toNumber());
+            i++;
+            if (i === 2) {
+              off();
+              resolve(true);
+            }
+          } catch (e) {
+            off();
+            reject(e);
+          }
+        });
+      });
+      const success = (await assertBlocks) as any;
+      assert.equal(success, true, success);
+      // assert that second block's timestamp is at least `blockTime` greater
+      // than the first block's. meaning, these blocks weren't mined one after
+      // the other
+      assert(timestamps[1] >= timestamps[0] + blockTime);
```

Comment:
I'm pretty sure there's no scenario where this can _not_ be the case right? 

The "wait" timer starts _after_ the first block is mined, so even under strange performance / load conditions this should be correct?

Reply:
I think you're right @jeffsmale90 . Though the check Is nice because the miner has options to mine many blocks while enforcing 1 transaction per block. So while someone would really have too break multiple things in the mine call to trigger this issue, it could still catch that sort of mistake. Maybe we should also check that the second block is < 2x the block time?

Reply:
@jeffsmale90 This test does not pass without the associated fix in the blockchain file. Previously, when the `blockTime` first elapsed, we'd mine _all transactions in the pool_, regardless of how many blocks it took. So in this test scenario, two blocks would be mined back to back.

The blockchain fix ensures that we only one mine one block every `blockTime`. This test confirms that fix. So, you're right that this should always pass. Unless we break something, then this will catch it :smile: 

Unless I'm misunderstanding you?

Reply:
I agree with the change - I was considering whether there is a scenario where this will fail due to inconsistent timing.

I believe that is _not_ the case, assuming the interval timer is started _after_ mining is complete, which I believe is the case.

ie. if the interval was created with `setInterval` rather than `setTimeout` _after_ the block is mined, this test would be inconsistent.